packages/edit-site/src/components/page-patterns/dataviews-patterns.js

```diff
@@ -235,12 +253,33 @@ export default function DataviewsPatterns() {
 				render: ( { item } ) => (
 					<Title item={ item } categoryId={ categoryId } />
 				),
-				maxWidth: 400,
 				enableHiding: false,
 			},
-		],
-		[ view.type, categoryId ]
-	);
+		];
+		if ( type === PATTERN_TYPES.theme ) {
```

Comment:
How do I trigger a situation in which the category type is `PATTERN_TYPES.user`?

Reply:
To test

- I went to the "manage my patterns" screen
- edited one of my patterns by adding a new "howdy" category

I see this new category is listed in the patterns page, though it works as any other:

https://github.com/WordPress/gutenberg/assets/583546/204b1c4b-5f0c-49cc-a8e5-718af1b0af93

Should I do anything differently?

Reply:
This is for later, but making `sync` visible and changing the listing hide the field (this is: the sidebar is not using the views yet, like the pages page).

https://github.com/WordPress/gutenberg/assets/583546/70496bfa-2388-4ebc-a26f-3d917e7468a6

Though, this makes me think that I wouldn't want the `sync` field to be removed from the view. Consider the following case: a user is navigating through the categories to see what patterns are available, though they are only interested in "not synced" patterns:

- `sync` field is shown in the layout
- `sync` filter is also shown

<img width="1010" alt="Captura de ecrã 2024-01-04, às 14 39 48" src="https://github.com/WordPress/gutenberg/assets/583546/e418f373-5c0b-496d-b99e-0a4470d8c386">

What happens when the user visits a user-provided category and then a theme provided category?

The consequences are that 1) visual jumps in the layout and 2) the user preferences (hiddenFields and filter statuses) are cleared once they visit a user-provided category.

It sounds it's to always render the `sync` field and its filter.

Reply:
> Did you encountered something unexpected that is different from the old patterns page?

I don't know, because I don't know how to trigger a situation where type is not `PATTERN_TYPES.thems` :)

From what I gathered, my suggestion here is to always render the `sync-status` field and its filter.

Reply:
We definitely don't want to render it when `categoryType=wp_template_part`. I just used the same logic for displaying the filter as in the old page.

Reply:
> Maybe @talldan or @aaronrobertshaw can shed some light here?

Originally, users couldn't create or manage their own pattern categories. Themes and core, however, added categories via `WP_Block_Pattern_Categories_Registry` and `register_block_pattern_category`.

To allow user-created patterns a taxonomy was used however it was a bit of an awkward fit in terms of maintaining backwards compatibility with the theme/core registered patterns. Eventually, those were able to be absorbed, so my current understanding is that on the patterns page they are essentially treated as the same thing now.

@glendaviesnz lead the efforts around pattern categorization and could provide more context or explanation if it's needed.

Reply:
Taking a look. What's confusing is there are two `PATTERN_TYPES` consts that seem to have different values:
https://github.com/WordPress/gutenberg/blob/6f61d8b802a880dc58eb490c17599cea0d614ba2/packages/block-editor/src/components/inserter/block-patterns-tab/utils.js#L7-L14

https://github.com/WordPress/gutenberg/blob/6f61d8b802a880dc58eb490c17599cea0d614ba2/packages/patterns/src/constants.js#L6-L9

Reply:
It seems the majority of the codebase uses the first variable, where `PATTERN_TYPES.user === 'user'`.

It seems you're using the second variable where `PATTERN_TYPES.user === 'wp_block'`, it doesn't seem to be used much in the codebase.

I think the first variable is the one you'll want to use, though I think it's still not great as the one object seems to mix and match lots of different concepts (syncing, pattern source, and I'm not sure what 'all' is).

This definitely needs some improvement!

Reply:
So I think sometimes we're working with one kind of pattern item (where type is `user`) and other times a different type of item (where type is `wp_block`).

I've started looking into improving the code quality. First PR here - [Rename patternBlock to patternPost](https://github.com/WordPress/gutenberg/pull/57568#top)

I think the goal should be to have a single normalized pattern item data structure (even if the data is sourced from different places) with consistent values, then hopefully we'll be able to tidy up the constants.

Reply:
I think I have a full understanding of what the code is doing now.

There's a `categoryType` sprinkled across the codebase, but this is purely for routing and can be one of two values, `pattern` (for any time of pattern) or `wp_template_part` (for template parts). It's confusing that `PATTERN_TYPES` is often used to check this in addition to being used to check the pattern item types. The category type values are also unusual with one being a post type and the other not.

I'll follow up with another PR that improves this, but it'll be next week.

Reply:
Nice, thanks for looking into this, Dan.